packages/http-client/src/http-request-config.ts

```diff
@@ -40,3 +46,90 @@ const defaultDisallowedKeys = [
   'headers',
   'params'
 ];
+
+/**
+ * Build [[HttpRequestConfig]] from [[HttpRequestConfigWithOrigin]]
+ * @param configWithOrigin - An http request config with origin information
+ * @returns The resulting [[HttpRequestConfig]]
+ * @internal
+ */
+export function buildHttpRequestConfig(
+  configWithOrigin: HttpRequestConfigWithOrigin
+): HttpRequestConfig {
+  const requestConfig = configWithOrigin as HttpRequestConfig;
+  requestConfig.headers = getOptionWithPriority(configWithOrigin.headers);
```

Comment:
I think the requirement is missing that we should merge ignoring the casing and keep the casing of the value which one. The `mergeLeftIgnoringCase` method should do it.

Reply:
- {'key': 'value'} (custom)
- {'KEY': 'value2'} (sdk)
=> {'key': 'value'}